ironfish-rust-nodejs/src/nacl.rs

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use ironfish_rust::nacl::{box_message, bytes_to_secret_key, new_secret_key, unbox_message};
+use ironfish_rust::{base64, hex, nacl};
+use napi::bindgen_prelude::*;
+use napi_derive::napi;
+
+#[napi]
+pub const KEY_LENGTH: u32 = nacl::KEY_LENGTH as u32;
+
+#[napi]
+pub const NONCE_LENGTH: u32 = nacl::NONCE_LENGTH as u32;
+
+#[napi]
+pub struct BoxKeyPair {
+    public_key: Vec<u8>,
+    secret_key: Vec<u8>,
+}
+
+#[napi]
+impl BoxKeyPair {
+    #[napi(constructor)]
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> BoxKeyPair {
+        let secret_key = new_secret_key();
+
+        BoxKeyPair {
+            public_key: secret_key.public_key().as_bytes().to_vec(),
+            secret_key: secret_key.as_bytes().to_vec(),
+        }
+    }
+
+    #[napi(factory)]
+    pub fn from_hex(secret_hex: String) -> napi::Result<BoxKeyPair> {
+        let byte_vec = hex::decode(secret_hex)
+            .map_err(|_| Error::from_reason("Unable to decode secret key".to_owned()))?;
+
+        let bytes: [u8; nacl::KEY_LENGTH] = byte_vec
+            .try_into()
+            .map_err(|_| Error::from_reason("Unable to convert secret key".to_owned()))?;
+
+        let secret_key = bytes_to_secret_key(bytes);
+
+        Ok(BoxKeyPair {
+            public_key: secret_key.public_key().as_bytes().to_vec(),
+            secret_key: secret_key.as_bytes().to_vec(),
+        })
+    }
+
+    #[napi(getter)]
+    pub fn public_key(&self) -> Uint8Array {
```

Comment:
Would it work to return `Buffer` everywhere around here instead of `Uint8Array`? I think most of our code is returning those right now.

Reply:
Yeah. I went with `Uint8Array` originally since that's what `tweetnacl` was returning, but I like just using `Buffer` since it's our standard anyway.